internal/app/cache/cache.go

```diff
@@ -11,9 +17,130 @@ type Cache interface {
 	// Fetch returns the cached response if it exists.
 	Fetch(key string) (*envoy_api_v2.DiscoveryResponse, error)
 
-	// SetResponse sets the cache response and returns the list of open watches.
+	// SetResponse sets the cache response and returns the list of requests.
 	SetResponse(key string, resp envoy_api_v2.DiscoveryResponse) ([]*envoy_api_v2.DiscoveryRequest, error)
 
-	// AddWatch adds the watch to the cache and returns whether a stream is open.
-	AddWatch(key string, req envoy_api_v2.DiscoveryRequest) (bool, error)
+	// AddRequest adds the request to the cache and returns whether a stream is open.
+	AddRequest(key string, req envoy_api_v2.DiscoveryRequest) (bool, error)
+}
+
+type cache struct {
+	cacheMu       sync.RWMutex
+	cache         *ristretto.Cache
+	expireSeconds time.Duration
+}
+
+type resource struct {
+	resp       *envoy_api_v2.DiscoveryResponse
+	requests   []*envoy_api_v2.DiscoveryRequest
+	streamOpen bool
+}
+
+func NewCache(numCounters int64, cacheSizeBytes int64, expireSeconds int, onEvict func(key, conflict uint64,
```

Comment:
thoughts on pulling out `onEvict` into its own type? Would make easier to document what some of the variables (key, conflict, value, cost) mean.

```
// onEvict ...
type onEvict func(uint64, uint64, interface{}, int64)
```

Reply:
Done. The documentation is a bit lacking on the first two parameters, but tldr we lose what the original value of the key is when this is called since we only passed in the hashed key.